File: slogen.php
#!/usr/bin/env php
<?php

function _usage() {
  echo "usage: slogen.php filename.txt\n\n";
  echo "options:\n";
  echo "--hashtag  // add #\n";
  echo "--twitterlimit // limits to 140 chars\n";
  echo "\n";
}

if (
     (in_array("--help", $argv)) || (in_array("-h", $argv))
   ) {
  _usage();
  exit();
}

if (!isset($argv[1])) {
  _usage();
  exit(1);
} else {
  if (!is_file($argv[1])) {
    echo "error: file '" . $argv[1] . "' not found\n";
    exit(1);
  }
  $txt = file_get_contents($argv[1]);
}
$txt = str_replace("\r", "", $txt);
$txt = trim($txt);

$chunks = explode("\n\n", $txt);
$parts = array();

foreach($chunks as $chunk) {
  $words = explode("\n", $chunk);
  $parts[] = $words[rand(0, count($words)-1)];
}

$output = implode(' ', $parts);
if (in_array("--twitterlimit", $argv)) {
  if (strlen($output) > 140) {
    $output = "";
  }
}

if (in_array("--hashtag", $argv)) {
  $chunks = explode(" ", $output);
  for ($i=1;$i<count($chunks);$i++) {
    if (substr($chunks[$i], 0, 1) == strtoupper(substr($chunks[$i], 0, 1))) {
      $chunks[$i] = "\\#" . $chunks[$i];
    }
  }
  $output = implode(' ', $chunks);
}

echo $output;

File: slogen.py
#!/usr/bin/env python
import sys
from random import randint

try:
  filename = sys.argv[1]
#except FileNotFoundError:
#  print 'file not found\n'
except IndexError:
  print('usage: slogen.py filename.txt\n')
  exit(1)
except:
  print("Unexpected error:", sys.exc_info()[0])
  exit(1)

do_hashtag = False
if ('--hashtag' in sys.argv):
  do_hashtag = True

data = open(filename, 'r').read()
chunks = data.split('\n\n')

msg = ''
for chunk in chunks:
  lines = chunk.strip().split('\n')
  if len(lines) > 1:
    rnd = randint(0, len(lines)-1)
    msg = msg + ' ' +lines[rnd]

msg = msg.strip()
if do_hashtag:
  words = msg.split(' ')
  i = 0;
  for word in words:
    if ((word[0:1] == word[0:1].upper()) & (len(word) > 3)):
      word = '#' + word
      words[i] = word
    i = i + 1
  msg = ' '.join(words)

print(msg)
